apps/server/src/modules/files-storage/entity/filerecord.entity.ts

```diff
@@ -158,26 +158,76 @@ export class FileRecord extends BaseEntity {
 		this.deletedSince = props.deletedSince;
 	}
 
-	updateSecurityCheckStatus(status: ScanStatus, reason: string): void {
+	public updateSecurityCheckStatus(status: ScanStatus, reason: string): void {
 		this.securityCheck.status = status;
 		this.securityCheck.reason = reason;
 		this.securityCheck.updatedAt = new Date();
 		this.securityCheck.requestToken = undefined;
 	}
 
-	markForDelete(): void {
+	public setSecurityCheck(securityCheck: FileSecurityCheck): void {
+		if (securityCheck instanceof FileSecurityCheck) {
```

Comment:
Wieso dieser Check? Ist das hier ein spezielles Risiko, dass was falsches reinkommt? Ich frage, weil ich so einen Check sonst noch nicht im Code gesehen hab.

Reply:
Das ist etwas zwiespältig. Hätten wir kein TS, müssten solche checks überall sein. Jetzt haben wir TS und es wird immer weg gelassen. Allerdings zur Laufzeit gibt es diesen check nicht. Es ist allerdings etwas das wir im Code produzieren. Daher das was dort gesetzt wird überhaupt nicht in der Datenstruktur geschützt ist und Methoden kaput gehen sollte dort was falsches rein kommen. Das new wird an anderer Stelle ausgeführt. Vielleicht müsste man das verlagern. Ich bin mir da nicht soo sicher.